### PR TITLE
chore: a11y lint plugin

### PR DESCRIPTION
**Pull request by ragi96**
Activated `eslint-plugin-jsx-a11y` on all blocks

disabled all errors on color, color-kit, checklist and color-scale as they need to be overhauled anyway before they are finished.
disabled all errors on the gradient-block as there is already worked on it https://github.com/Frontify/guideline-blocks/pull/609

Fixed a lot of the errors.

### CODE REVIEW

File: examples/asset-upload/.eslintrc.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /* (c) Copyright Frontify Ltd., all rights reserved. */
 
 module.exports = {
-    extends: ['@frontify/eslint-config-react'],
+    extends: ['@frontify/eslint-config-react', 'plugin:jsx-a11y/recommended'],
```
Comment (SamuelAlev) [inline examples/asset-upload/.eslintrc.js:4]:
```suggestion
    extends: ['@frontify/eslint-config-react', 'plugin:jsx-a11y/strict'],
```
?

Comment (ragi96) [reply examples/asset-upload/.eslintrc.js:4]: Will have a look 👍

File: packages/audio-block/src/components/AudioPlayer.tsx [hunk 2]
```diff
@@ -51,6 +51,7 @@ export const AudioPlayer = ({
                 <LoadingCircle />
             </div>
         ) : (
+            // eslint-disable-next-line jsx-a11y/media-has-caption
```
Comment (fulopdaniel) [inline packages/audio-block/src/components/AudioPlayer.tsx:54]: Can we turn this rule off in the eslintrc?

File: packages/checklist-block/src/components/Checkbox.tsx [hunk 3]
```diff
@@ -53,6 +53,7 @@ export const Checkbox: FC<CheckboxProps> = ({
     };
 
     return (
+        // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-noninteractive-element-interactions
```
Comment (fulopdaniel) [inline packages/checklist-block/src/components/Checkbox.tsx:56]: Do you think it makes sense to add an onKeyDown listener for the "Enter" key that also calls `handleLabelClick` ?

File: packages/color-block/src/components/cards/CardsItem.tsx [hunk 4]
```diff
@@ -74,6 +74,7 @@ export const CardsItem = ({ color, colorSpaces, isEditing, onBlur, onUpdate, onD
                     }
                     triggerElement={
                         <div className="tw-overflow-hidden tw-rounded-t tw-bg-[url('https://cdn.frontify.com/img/transparent.png')] tw-bg-[length:10px_10px]">
+                            {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions */}
```
Comment (fulopdaniel) [inline packages/color-block/src/components/cards/CardsItem.tsx:77]: add role="button" and a tabIndex={0} to overcome the second rule. For the first rule, maybe it's worth to add an onkeydown listener to Enter here as well.

File: packages/color-block/src/components/cards/CardsItem.tsx [hunk 5]
```diff
@@ -138,6 +139,7 @@ export const CardsItem = ({ color, colorSpaces, isEditing, onBlur, onUpdate, onD
                                                 />
                                             }
                                             triggerElement={
+                                                // eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions
```
Comment (fulopdaniel) [inline packages/color-block/src/components/cards/CardsItem.tsx:142]: Same as above.

File: packages/checklist-block/src/components/Checkbox.tsx (hunk 3 above)
Comment (ragi96) [reply packages/checklist-block/src/components/Checkbox.tsx:56]: Won't tackle as its an unreleased block and needs some work anyway before go live

File: packages/color-block/src/components/cards/CardsItem.tsx (hunk 4 above)
Comment (ragi96) [reply packages/color-block/src/components/cards/CardsItem.tsx:77]: Won't tackle as its an unreleased block and needs some work anyway before go live

File: packages/color-block/src/components/cards/CardsItem.tsx (hunk 5 above)
Comment (ragi96) [reply packages/color-block/src/components/cards/CardsItem.tsx:142]: Won't tackle as its an unreleased block and needs some work anyway before go live

File: packages/figma-block/src/FigmaBlock.tsx [hunk 6]
```diff
@@ -103,6 +103,7 @@ export const FigmaBlock = ({ appBridge }: BlockProps): ReactElement => {
     };
 
     const FigmaEmptyBlock = () => (
+        // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
```
Comment (ragi96) [inline packages/figma-block/src/FigmaBlock.tsx:106]: Fix this one

File: packages/audio-block/src/components/AudioPlayer.tsx (hunk 2 above)
Comment (ragi96) [reply packages/audio-block/src/components/AudioPlayer.tsx:54]: In my opinion this rule has a point and we should check with martin how we wanna implement the option to add caption tracks in the audio block and later in the video block

File: packages/figma-block/src/FigmaBlock.tsx (hunk 6 above)
Comment (ragi96) [reply packages/figma-block/src/FigmaBlock.tsx]: Nice one @ragi96

File: examples/asset-upload/.eslintrc.js (hunk 1 above)
Comment (SamuelAlev) [reply examples/asset-upload/.eslintrc.js:4]: They want you to think it is enabled 😄 

Screenshot 2023-09-08 at 15 48 58

**Review by fulopdaniel — APPROVED** [review]

**Comment by ragi96** [comment]
Will merge, if @LidijaStoilova recommends to go with strict, i will open a follow up PR

**Pull request merged**